test/edu-benefits/1995/config/contactInformation.unit.spec.jsx

```diff
@@ -0,0 +1,130 @@
+      ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'select')
+    );
+
+    expect(inputs.filter(input => input.id.startsWith('root_preferredContactMethod')).length)
+      .to.equal(3);
+    expect(inputs.filter(input => input.id.startsWith('root_veteranAddress')).length)
+      .to.equal(6);
+    expect(inputs.filter(input => input.id.startsWith('root_view:otherContactInfo')).length)
+      .to.equal(4);
+  });
+  it('should render required fields', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+
+    const formDOM = findDOMNode(form);
+    ReactTestUtils.findRenderedComponentWithType(form, Form).onSubmit({
+      preventDefault: f => f
+    });
+
+    expect(formDOM.querySelectorAll('.usa-input-error').length).to.equal(6);
+  });
+  it('should not render confirm email on review page', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{
+            'view:otherContactInfo': {
+              'view:confirmEmail': 'test@test.com'
+            }
+          }}
+          reviewMode
+          uiSchema={uiSchema}/>
+    );
+
+    expect(findDOMNode(form).textContent).not.to.contain('test@test.com');
+  });
+  it('should conditionally require phone number', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          formData={{}}
+          schema={schema}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+
+    const formDOM = findDOMNode(form);
+
+    ReactTestUtils.findRenderedComponentWithType(form, Form).onSubmit({
+      preventDefault: f => f
+    });
+
+    // jsdom has issues with colons in attributes
+    let errors = Array.from(formDOM.querySelectorAll('.usa-input-error > label'));
+    let phoneError = errors.find(errorLabel => errorLabel.getAttribute('for').endsWith('homePhone'));
+    expect(phoneError).to.be.undefined;
+
+    const phoneMethod = ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'input')
+      .find(input => input.getAttribute('id') === 'root_preferredContactMethod_2');
+    ReactTestUtils.Simulate.change(phoneMethod, {
+      target: {
+        value: 'phone'
+      }
+    });
+
+    errors = Array.from(formDOM.querySelectorAll('.usa-input-error > label'));
+    phoneError = errors.find(errorLabel => errorLabel.getAttribute('for').endsWith('homePhone'));
+    expect(phoneError).not.to.be.undefined;
+  });
+  it('should show error if emails do not match', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          formData={{}}
+          schema={schema}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+
+    const formDOM = findDOMNode(form);
+    const inputs = ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'input');
+    const emailInput = inputs.find(input => input.id.endsWith('email'));
```

Comment:
Is this case-sensitive? i.e., is it guaranteed to find the email input and not the confirmationEmail input (may work even if case-insensitive since email is first on the page, but that would be kind of brittle)